tracer/src/Datadog.Trace/Iast/SensitiveData/EvidenceRedactor.cs

```diff
@@ -81,9 +83,9 @@ public bool IsValueSensitive(string? value)
         {
             return _valuesRegex.IsMatch(value);
         }
-        catch (Exception err)
+        catch (RegexMatchTimeoutException err)
         {
-            _logger?.Error("Timeout in Evidence Values Redaction Regex. {V}", err.ToString());
+            _logger?.Error(err, "Regex timed out when trying to match value {Input} against pattern {Pattern}.", err.Input, err.Pattern);
```

Comment:
This almost certainly shouldn't be an error IMO. We could be running this method hundreds of thousands of times a second in a high-throughput web app, and if it times out once, it's _likely_ to time out a _lot_, which means we going to be _flooding_ their logs. The only "safe" value to use here is `Debug`. 

We could potentially add a simple bool that checks if we've recorded a timeout before (or potentially, a timeout for _this pattern_). If we haven't, write an `Error` log (or potentially a `Warning`, which seems more appropriate?), and then set the bool. On subsequent timeouts, write a debug log instead.

We also _mustn't_ write the input value to the logs IMO. By it's nature, this could contain sensitive data, and we should never explicitly leak sensitive data in the logs like this.

Reply:
I was thinking about it... I finally followed the same approach as other existing RegexMatchTimeoutExceptions (We should review that as well). Still, I think that it's a good idea to show a warning in the first error and then only debug traces, so I have updated the code. I've removed the input value. Thanks!